tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/LockFreeExponentiallyDecayingReservoirWithSamples.java

```diff
@@ -79,34 +77,43 @@ private static final class State {
         private final int size;
         private final long startTick;
         // Count is updated after samples are successfully added to the map.
-        private final ConcurrentSkipListMap<Double, WeightedSample> values;
+        private final ConcurrentSkipListMap<Double, WeightedSampleMetadata> values;
```

Comment:
we should duplicate this class into a new one to handle exemplars, with a name that fits better. We can do that later on, the way this is structured makes the diff easier to follow for now :-)

Reply:
You mean the reservoir class or the WeightedSampleMetadata? I can def rename WeightedSampleMetadata to something like WeightedSampleWithExemplar.

Reply:
The reservoir type, sorry I was ambiguous! For later though, since this makes it easier to review for the time being